tests/common/include/tests/common/tools/create_toy_geometry.hpp

```diff
@@ -164,6 +170,10 @@ void create_cyl_volume(detector_t &det, vecmem::memory_resource &resource,
                      inner_r, outer_r, upper_z, edges[3]);
 
     det.add_objects(ctx, cyl_volume, surfaces, masks, transforms);
+    if (cyl_volume.get_grid_type() !=
+        detector_t::volume_type::grid_type::e_no_grid) {
+        det.add_surfaces_grid(ctx, cyl_volume, cyl_surfaces_grid);
```

Comment:
As far as I can see, portal surfaces need to be added to the grid, too, if they should be found by the navigator

Reply:
They should be found by the navigator but I don't think they have to be added to the grid. 
I think the candidate surfaces are found from locally resolved surfaces (grid) + portals

Reply:
Then you will have to rewrite the navigator, I'm afraid. What is your reason to keep the portals out of the grid?

Reply:
Nvm, I think I see where it might get difficult. We might have to reintroduce the portal range then...

Reply:
AFAIK, the navigator is missing local navigation function - so some parts need to be rewritten or enhanced anyway. 

> What is your reason to keep the portals out of the grid?

1. I don't have any good reason put them in the grid
2. I don't know how to clearly define z(r) - phi value of portal surfaces that would be fed into grids of surface finders
2. The number of portals per volume is quite few so we don't lose many time for resolving them

Reply:
> We might have to reintroduce the portal range then...

Is that so? hmm...

Reply:
Fastest way would be adding a portal range in `volume` class - I know it is a bit dirty but should work

Reply:
No, put the portal range back into the volume is what I meant actually. We will have to adjust the detector though, but maybe it is good to able to resolve the portal indices anyways.

Reply:
I think you are right, the navigator doesn't need much attention, if we just let the surfaces finder return all of the portals by default, together with the module surface neighborhood. That's safer against getting trapped in a volume, too.